shuriken-dump/shuriken-dump.cpp

```diff
@@ -439,4 +452,53 @@ void print_code(std::span<std::uint8_t> bytecode) {
         fmt::print(" Flags:                 0x{:X}\n", macho_header.flags);
         fmt::print(" Reserved:              0x{:X}\n", macho_header.reserved);
     }
+
+    void print_macho_loadcommands(const shuriken::parser::macho::MachoCommands &commands) {
+        const shuriken::parser::macho::MachoCommands::loadcommands_t& macho_loadcommands = commands.get_macho_loadcommands_const();
+
+        for (const std::unique_ptr<shuriken::parser::macho::MachoCommands::loadcommand_t>& loadcommand : macho_loadcommands) {
+            fmt::println("Load command:\n");
+            fmt::print(" Type:                  0x{:X}\n", loadcommand->cmd);
+            fmt::print(" Size:                  {} bytes\n", loadcommand->cmdsize);
+        }
+    }
+
+    void print_macho_segmentcommands(const shuriken::parser::macho::MachoCommands &commands) {
+        const shuriken::parser::macho::MachoCommands::segmentcommands_t& macho_segmentcommands = commands.get_macho_segmentcommands_const();
+
+        for (const std::unique_ptr<shuriken::parser::macho::MachoCommands::segmentcommand_t>& segmentcommand : macho_segmentcommands) {
```

Comment:
If you replace keep a cache vector of `reference_wrapper`, you will avoid returning unique_ptr, keeping the ownership in the correct object. Check how we do it in the other parts.